### PR TITLE
Contact modal and form

### PR DESCRIPTION
**Pull request by PaulinaMurzanskaSofomo**
This is a contact form component. Very large and complex. 
The form itself (ContactFormik) is managed by Formik, as well as Inputs. 

The ContactFormik is displayed as children by the ContactModal component, which gives all the layout and animations. This component has two scenarios:

   1. Initially hidden with a negative z-index and fixed to the screen, open as a pop-up when the user clicks on the trigger ( mainly contact buttons).
   
  3. Visible on the Outsource page ( not created yet ). It will be visible and fixed to the page when the screen will be wider than mobile, and hidden on the mobile screen.

I use context to manage the state of the modal, state change will be triggered within many components, so I figured it will be a better solution than lifting up the state. 

The form on submit will console log the data, and reset the form.
There is error handling implemented with Yup, as suggested in Formik docs.

I fixed the position of the modal with a negative z-index, to achieve an animation effect, not the initial one when the modal scale and fade up, but the reverse one, when we close it, it scales down. All other solutions I tried, made the modal to close immediately without this scale-down effect.

All those components created with this PR, are strictly connected to each other, so I figured it will make sense if I create them altogether, to see what's going on with the whole functionality.

### CODE REVIEW

File: src/organisms/ContactFormik/ContactFormik.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <StyledContactForm>
+      <Formik
+        validateOnChange={true}
+        initialValues={{
+          name: "",
+          email: "",
+          subject: "",
+          message: "",
+        }}
```
Comment (patrykJankowiak1) [inline src/organisms/ContactFormik/ContactFormik.tsx:24]: lets move initialValues object outside

**Review by patrykJankowiak1 — COMMENTED** [review]
left few comments, please don't merge before you get approval

**Comment by PaulinaMurzanskaSofomo** [comment]
Ok, made all the changes, please check the result when you have time:)

**Comment by patrykJankowiak1** [comment]
looks good. my only comment would be that this hook doesn't necessarily need to separated from the context.

other than that, good to go

**Review by patrykJankowiak1 — APPROVED** [review]

**Pull request merged**